src/v/features/feature_table.cc

```diff
@@ -697,6 +706,49 @@ void feature_table::assert_compatible_version(bool override) {
     }
 }
 
+feature_table::probe::probe(const feature_table& parent)
```

Comment:
`feature_table` can be passed through `setup_metrics`.

Reply:
I've left it to be passed in the constructor because then I can hold it as a reference instead of a pointer inside the probe which means that I don't have to deal with a `nullptr`. tldr: it's simpler to pass it in the constructor and `setup_metrics` gets called right after the constructor so there's not much of a difference here

Reply:
I thinking to capture it directly into the lambda:
```c++
             [&ft]() {
                  return calculate_expiry_metric(ft.get_license());
              },
```
But it's a minor point.